invenio_rdm_records/records/systemfields/access/field.py

```diff
@@ -107,24 +105,23 @@ def errors(self):
 
     @embargo.setter
     def embargo(self, embargo):
-        self._embargo = embargo
+        if embargo is None:
+            embargo = Embargo.NONE
 
-        if embargo is not None:
-            # allow 'clear_embargo()' to be called as 'embargo.clear()'
-            self._embargo.clear = lambda: self.clear_embargo()
+        # allow 'clear_embargo()' to be called as 'embargo.clear()'
+        self._embargo = embargo
+        self._embargo.clear = lambda: self.clear_embargo()
```

Comment:
true, that strange construct above was only there because from within the Embargo instance, we couldn't set the reference to itself, that the Access object has, to None...
now that we did away with the None variant, it's not required anymore